csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DotnetCommand.cs

```diff
@@ -0,0 +1,61 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using Semmle.Util;
+
+namespace Semmle.Extraction.CSharp.DependencyFetching
+{
+    /// <summary>
+    /// Low level utilities to run the "dotnet" command.
+    /// </summary>
+    internal sealed class DotnetCommand : IDotnetCommand
+    {
+        private readonly ProgressMonitor progressMonitor;
+
+        public string Exec { get; }
+
+        public DotnetCommand(ProgressMonitor progressMonitor, string exec)
+        {
+            this.progressMonitor = progressMonitor;
+            this.Exec = exec;
+        }
+
+        private ProcessStartInfo MakeDotnetStartInfo(string args, bool redirectStandardOutput)
+        {
+            var startInfo = new ProcessStartInfo(Exec, args)
+            {
+                UseShellExecute = false,
+                RedirectStandardOutput = redirectStandardOutput
+            };
+            // Set the .NET CLI language to English to avoid localized output.
+            startInfo.EnvironmentVariables["DOTNET_CLI_UI_LANGUAGE"] = "en";
+            return startInfo;
+        }
+
+        public bool RunCommand(string args)
+        {
+            progressMonitor.RunningProcess($"{Exec} {args}");
+            using var proc = Process.Start(MakeDotnetStartInfo(args, redirectStandardOutput: false));
```

Comment:
What happens if we have a `ProcessStartInfo` with `redirectStandardOutput = false`, and we call `ReadOutput(out output)` on it? 

It feels odd that we can't share more logic between the two `RunCommand` variants.

Reply:
Then it will crash. I will make `ReadOutput` more robust (which seems sensible enough).
The other option is to make a ReadCommand wrapper method for logging, but it will quickly be a bit nasty as it requires and extra level of indirection when using output variables in lambdas.